src/libraries/System.Text.Json/gen/System.Text.Json.SourceGeneration.Roslyn4.4.csproj

```diff
@@ -17,4 +20,20 @@
     <Compile Include="$(CoreLibSharedDir)System\Collections\Generic\ValueListBuilder.Pop.cs" Link="Production\ValueListBuilder.Pop.cs" />
   </ItemGroup>
 
+  <ItemGroup Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">
```

Comment:
I see what you mean.  This creates an interesting layering.  Suppose we wanted to do this for interop generator.  That generator itself would need to be referenced by the source assemblies.  I guess it would work since you're only referencing the ref projects...  I'm not sure we "care" about granular references for source-generators.  They aren't sitting at some layer within the framework.  They're design time components and are free to use the entire framework.

Instead of hand-selecting all these references, could we instead build the source generators after the references for netcoreappcurrent was built and let them consume all of the framework, similar to how OOB assemblies might?

Reply:
> Instead of hand-selecting all these references, could we instead build the source generators after the references for netcoreappcurrent was built and let them consume all of the framework, similar to how OOB assemblies might?

We eventually want to merge our reference and source projects and make the reference assembly output just a by-product of a project's compilation (either by using Roslyn's RefOut or by manually compiling reference source files). Without strongly defined dependencies in inbox source generators, we would need to keep the current reference assembly build phase instead of inlining them and that's something that I wanted to avoid (it requires double evaluations of projects and it creates an undesirable build sequence).

Reply:
After combining references this will become _much_ harder.  Imagine the case where a source generator needs a reference to an assembly (say System.IO) and that same assembly needs to use the source generator (imagine the DLLImport generator).  It cannot take that reference, and clearly we should opt for up-to-date source generator over up-to-date references in such a case.  These source generators should not be sitting in the library stack at all - they're design time components.

Maybe instead of using live references here we should use LKG and disable the configuration for source build (or perhaps disable in all but the allconfigurations build)?  The tradeoff there would be that the references would be a little stale but IMO that's an acceptable compromise.

Reply:
> Maybe instead of using live references here we should use LKG and disable the configuration for source build (or perhaps disable in all but the allconfigurations build)?

At the point when .NET Standard source generators go away, how would you image this to work in a source build environment that doesn't have access to a LKG (i.e. without bootstrapping)? That won't happen anytime soon but it's something to discuss before we go into a specific direction.

Reply:
Probably the same way it would work for the compiler.  In source build the compiler consumes an "LKG" of the runtime to use for targeting / testing / building.  We then build the runtime with that compiler, then once everything is done we rebuild the compiler with the new runtime.